.github/workflows/build_and_deploy_sphinx_docs.yaml

```diff
@@ -0,0 +1,34 @@
+name: Deploy Sphinx Docs to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+
+      - name: Install required sphinx deps
+        run: |
+          pip install .[sphinx_docs]
+
+      - name: Build Documentation
+        run: |
+          make build-sphinx
+
+      - name: Deploy to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@v4 # this will deploy the html file built to gh-pages
+        with:
+          branch: gh-pages # using a separate branch here to avoid building sphinx files onto main
```

Comment:
This is a branch off main that lives indefinitely ?  Any danger in it getting really far behind main?